airbyte-integrations/connectors/destination-bigquery-denormalized/src/main/java/io/airbyte/integrations/destination/bigquery/formatter/DefaultBigQueryDenormalizedRecordFormatter.java

```diff
@@ -112,14 +113,17 @@ private void populateEmptyArrays(final JsonNode node) {
   private void surroundArraysByObjects(final JsonNode node) {
     findArrays(node).forEach(
         jsonNode -> {
-          final JsonNode arrayNode = jsonNode.deepCopy();
+          if (isAirbyteArray(jsonNode.get(ARRAY_ITEMS_FIELD))) {
+            final ObjectNode arrayNode = jsonNode.get(ARRAY_ITEMS_FIELD).deepCopy();
+            final ObjectNode originalNode = (ObjectNode) jsonNode;
 
-          final ObjectNode newNode = (ObjectNode) jsonNode;
-          newNode.removeAll();
-          newNode.putArray(TYPE_FIELD).add("object");
-          newNode.putObject(PROPERTIES_FIELD).set(NESTED_ARRAY_FIELD, arrayNode);
+            originalNode.remove(ARRAY_ITEMS_FIELD);
+            final ObjectNode itemsNode = originalNode.putObject(ARRAY_ITEMS_FIELD);
```

Comment:
checking my understanding:
* previously, a JsonNode representing an array schema would be replaced with a JsonNode representing an object schema, containing a `big_query_array` property whose value was the original schema
* the new logic is: the outer JsonNode is not wrapped inside an object, but it the items inside that array are, themselves, arrays, then they will still be wrapped in objects?

if that's accurate: wouldn't this have the same problem? Why do we need to do this wrapping at all?

https://github.com/airbytehq/airbyte/issues/11109 sounds like the user wants to recreate the source schema as closely as possible - this seems like we're basically pushing the problematic behavior down one level.

Reply:
BigQuery has limitations for arrays. You can't create an array of arrays.
![image](https://user-images.githubusercontent.com/30464745/179348042-10f13b74-9fda-4e52-9475-40d7eb5e8f37.png)
To avoid failure old implementation replaces all arrays by objects with the array field. 
As result, `Array of arrays` becomes `Object with array of objects with array`.
The new implementation does the schema transformation only when it's needed. We find cases when we have `array of arrays` and replace it with `array of objects with array`.

So, we change the incoming schema only when it's required.